bundles/org.openhab.binding.upnpcontrol/src/main/java/org/openhab/binding/upnpcontrol/internal/handler/UpnpRendererHandler.java

```diff
@@ -1113,6 +1113,11 @@ private void onValueReceivedVolume(String variable, @Nullable String value) {
             long volume = Long.valueOf(value);
             volume = volume * 100 / config.maxvolume;
 
+            if (volume < 0) {
```

Comment:
Perhaps this could be checked right after line 1113 to avoid performing any calculations with invalid value?

Reply:
In the identified error case, volume came from the service as a negative. But there might also be a case where config.maxvolume (also retrieved from the service) is negative and that would lead to problems as well. Or we catch this bad data where config.maxvolume is set as well, or we leave this statement where it is.

Reply:
Too many ORs 😅 So what do we wanna do?

Reply:
Move it right after line 1113, but also change here: https://github.com/openhab/openhab-addons/blob/e2571d6f09984c9d0154eb0b54ea1fc4a7a630be/bundles/org.openhab.binding.upnpcontrol/src/main/java/org/openhab/binding/upnpcontrol/internal/services/UpnpRenderingControlConfiguration.java#L55
Put the following on that line:
```
if (volume && volume > 0)
```
In principle, it should be a positive integer less than or equal to 100. You could also log a warning there if it is not.